RecyclerView/app/src/main/res/layout/dialog_add_state.xml

```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:orientation="vertical" android:layout_width="match_parent"
+    android:layout_height="match_parent">
+
+    <EditText
+        android:id="@+id/stateName"
```

Comment:
this is ok, but it is common to apply snake case naming convention for xml ids maybe